SplunkPy: added `Fetch Investigations` (#44242)
demisto/content: Packs/SplunkPy/Integrations/SplunkPyV2/SplunkPyV2_description.md

***
# Splunk Enterprise Security Users

## Fetching events
The integration allows for fetching Splunk **Finding** and **Investigation** events using default queries. The queries can be changed and modified to support different Splunk use cases.
Palo Alto highly recommends reading the [Ingest Incidents from a SIEM Using Splunk article](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSOAR/6.x/Cortex-XSOAR-Tutorials-6.x/Ingest-Incidents-from-a-SIEM-Using-Splunk) before starting to use this integration.

> This article will help you configure your Splunk v2 integration, set up a basic flow, and start ingesting incidents from Splunk to Cortex XSOAR/XSIAM.

Reviewer: @ilappe this use case only explains ingesting incidents from Splunk to Cortex XSOAR.
It should be conditionalized for XSOAR only

Author: Done — wrapped the XSOAR tutorial paragraph in `<~XSOAR>...</~XSOAR>` and added a parallel `<~XSIAM>` paragraph (no link, since no XSIAM-side tutorial exists yet).

### How to configure
1. Navigate to one of the following:
    * Cortex XSOAR 8: Settings & Info > Settings > Integrations > Instances > SplunkPy v2
    * Cortex XSOAR 6: Settings > Integrations > Servers & Services > SplunkPy v2.

>     * Cortex XSIAM: Settings > Configurations > Automation & Feed Integrations > SplunkPy v2

Reviewer: ```suggestion
    * Cortex XSIAM: Settings > Data Sources & Integrations > SplunkPy v2
```

Author: Done.

2. Click **Add Instance**.
3. Type the **Server URL** and **Splunk Token**.
4. Select **Fetches incidents**.

> 5. Under Classifier, select **SplunkPyV2 - Classifier**. The classifier routes Finding events to the **Splunk Finding** incident type and Investigation events to the **Splunk Investigation** incident type.

Reviewer: @ilappe need to conditionalize for XSIAM issue type

Author: No need — the platform has a build-time process that swaps "incident type" → "issue type" for the XSIAM marketplace, so the source text stays as "incident type".

6. Under Mapper (incoming), select **Splunk ES - Incoming Mapper**.
7. Under Mapper (outgoing), select **Splunk ES - Outgoing Mapper**.

> 8. In the **Event types to fetch** parameter, select which Splunk event types to ingest. **Finding** is selected by default. Add **Investigation** to additionally fetch Splunk investigations (you can keep both selected to fetch them together).

Reviewer: ```suggestion
8. In the **Event types to fetch** parameter, select which Splunk event types to ingest. Default is **Finding**. Add **Investigation** to also fetch Splunk investigations (select both to fetch them together).
```

Author: Done.

> 9. Keep the **Fetch events query** as is, as we use the \`notable\` macro when ingesting Finding events. You can create a more granular search by specifying specific conditions such as specific security domain, event ID, etc.

Reviewer: ```suggestion
9. Keep the **Fetch events query** as-is, since we use the \`notable\` macro when ingesting Finding events. You can create a more granular search by specifying specific conditions, for example specific security domain or event ID.
```

Author: Done.

10. Keep the default values for the Finding **fetch limit** and **First fetch timestamp**.
11. (Investigation only) If **Investigation** is selected in step 8, configure the following Investigation-specific parameters:

>     * **First fetch timestamp (Investigations)** — the backfill window for the first investigations fetch (e.g., `7 days`).

Reviewer: ```suggestion
    * **First fetch timestamp (Investigations)**: The relative time interval to look back during the initial investigation fetch (for example, `7 days`).
```

Author: Done.

>     * **Maximum investigations per fetch** — the per-cycle limit (capped at 100 by Splunk).

Reviewer: ```suggestion
    * **Maximum investigations per fetch**: The maximum number of investigations to fetch per cycle. Limited to 100 by the Splunk investigations endpoint.
```

Author: Done.

>     * **Investigations fetch query** — keep as-is unless you need to filter further. The query must contain the literal token `FETCH_FILTER_PLACEHOLDER` — **do not modify or remove this token**. The integration replaces it at runtime with the time-range and pagination filters required to manage the fetch cycle. To add custom filters, append extra URL parameters such as `&status=New` alongside the placeholder.

Reviewer: ```suggestion
    * **Investigations fetch query**: Keep the SPL query as-is unless you need to filter further. The query must include the `FETCH_FILTER_PLACEHOLDER` token. Do not modify or remove this token. To add custom filters, append extra URL parameters such as `&status=New` after the placeholder.
```

Author: Done.

> 12. To enable mirroring in both environments, in the Incident Mirroring Direction field, select **Incoming and Outgoing**.

Reviewer: ```suggestion
12. To enable mirroring in both environments, in the **Incident Mirroring Direction** field, select **Incoming and Outgoing**.
```

Author: Done.

> Outgoing mirroring is recommended for Cortex XSOAR version 6.2 and above.

Reviewer: Should be conditionalized for XSOAR

Author: Done.

> 13. Select *Close Mirrored XSOAR Incident* and *Close Mirrored Splunk Finding Event* so that closing an incident in one environment automatically closes it in the other.

Author: Done

> 13. Select *Close Mirrored XSOAR Incident* and *Close Mirrored Splunk Finding Event* so that closing an incident in one environment automatically closes it in the other.

Reviewer: Should be conditionalized for XSOAR if relevant

14. In the Enrichment Types field, select *Asset*, *Drilldown* and *Identity*.
This enrichment provides additional information about assets, drilldown, and identities that are related to the finding events you ingest.
For more information, see [Enriching Finding Events](#enriching-finding-events).

> 15. Fetch backwards window - this backward window is for cases where there is a gap between the event occurrence time and the event index time on the Splunk server.

Reviewer: ```suggestion
15. Fetch backwards window - the time overlap for each fetch to account for the gap between event creation and indexing on Splunk. This helps prevent data gaps caused by ingestion lag.
```

Author: Done.
